File: dailyQuestion/2020-06/06-13/python/solution.py
def groupAnagrams(strs: [str]) -> [[str]]:
    if not strs :
        return [[]]
    result_tmp = {}
    for item in strs :
        tmp = "".join(sorted(item))
        if tmp in result_tmp :
            result_tmp[tmp].append(item)
        else :
            result_tmp[tmp] = [item]
    result = list(result_tmp.values())
    return result

if __name__ == "__main__" :
    strs = ["eat","tea","tan","ate","nat","bat"]
    result = groupAnagrams(strs)
    print(result)

File: dailyQuestion/2020-06/06-13/golang/solution.go
package main

import (
	"fmt"
	"strings"
)
import "sort"

func main() {
	strs := []string{"eat", "tea", "tan", "ate", "nat", "bat"}
	result := groupAnagrams(strs)
	fmt.Println(result)

}
func groupAnagrams(strs []string) [][]string {
	result := [][]string{}
	result_tmp := map[string][]string{}
	for _, item := range strs {
		tmp := strings.Split(item, "")
		sort.Strings(tmp)
		key := strings.Join(tmp, "")
		if _, ok := result_tmp[strings.Join(tmp, "")]; ok {
			result_tmp[key] = append(result_tmp[key], item)
		} else {
			result_tmp[key] = []string{item}
		}
	}
	for _, v := range result_tmp {
		result = append(result, v)
	}
	return result
}
